src/gazebo_gps_plugin.cpp

```diff
@@ -26,49 +26,67 @@
 
 #include <gazebo_gps_plugin.h>
 
+#include <boost/algorithm/string.hpp>
+
+using namespace std;
+
 namespace gazebo {
-GZ_REGISTER_MODEL_PLUGIN(GpsPlugin)
+GZ_REGISTER_SENSOR_PLUGIN(GpsPlugin)
 
-GpsPlugin::GpsPlugin() : ModelPlugin()
+GpsPlugin::GpsPlugin()
```

Comment:
Maybe I don't understand the architecture properly, but why are we not inheriting from the `SensorPlugin()` class?

Reply:
We are inheriting (check the header file) - what we aren't doing is initialising the base class, which from my checks, doesn't seem to be required.

Reply:
For all that matters, let me just still do it, as it seems more correct.